migrations/archiveMigration.js

```diff
@@ -0,0 +1,185 @@
+      await createTable(ARCHIVE_TABLE_NAME);
+    } else {
+      throw error;
+    }
+  }
+}
+
+async function restoreData(data) {
+  console.log('Processing data:', data.Items.length);
+
+  if (data.Items && data.Items.length > 0) {
+    console.log('Sending data to DynamoDB at endpoint:', dynamoClient.config.endpoint);
+
+    const BATCH_SIZE = 25;
+
+    for (let i = 0; i < data.Items.length; i += BATCH_SIZE) {
+      const batch = data.Items.slice(i, i + BATCH_SIZE);
+
+      // Prepare BatchWriteItem for PutRequest to archive table (write operation)
+      const batchWriteParams = {
+        RequestItems: {
+          [ARCHIVE_TABLE_NAME]: batch.map((item) => ({
+            PutRequest: {
+              Item: item
+            }
+          }))
+        }
+      };
+
+      // Prepare BatchWriteItem for DeleteRequest to original table (delete operation)
+      const batchDeleteParams = {
+        RequestItems: {
+          [TABLE_NAME]: batch.map((item) => ({
+            DeleteRequest: {
+              Key: {
+                pk: item.pk,
+                sk: item.sk
+              }
+            }
+          }))
+        }
+      };
+
+      try {
+        // Write items to the archive table
+        const result = await dynamoClient.send(new BatchWriteItemCommand(batchWriteParams));
+
+        if (result.UnprocessedItems && result.UnprocessedItems[ARCHIVE_TABLE_NAME]) {
+          console.warn(`Unprocessed write items: ${JSON.stringify(result.UnprocessedItems[ARCHIVE_TABLE_NAME])}`);
+        }
+
+        // Delete items from the original table
+        const deletedData = await dynamoClient.send(new BatchWriteItemCommand(batchDeleteParams));
+
+        if (deletedData.UnprocessedItems && deletedData.UnprocessedItems[TABLE_NAME]) {
+          console.warn(`Unprocessed delete items: ${JSON.stringify(deletedData.UnprocessedItems[TABLE_NAME])}`);
+        }
+
+        // Log progress
+        const totalBatches = Math.ceil(data.Items.length / BATCH_SIZE);
+        const batchesLeft = totalBatches - (Math.floor(i / BATCH_SIZE) + 1);
+        process.stdout.write(
+          `Batches written: ${Math.floor(i / BATCH_SIZE) + 1}/${totalBatches}. Total items written: ${i + batch.length}/${data.Items.length}. Batches left: ${batchesLeft}\r`
+        );
+      } catch (error) {
+        console.error('Error writing batch:', error);
+        throw error;
+      }
+    }
+    process.stdout.write('\r\x1b[K');
+  }
+}
+async function createTable(NEW_TABLE) {
+  const createTableParams = {
+    TableName: NEW_TABLE,
+    KeySchema: [
+      { AttributeName: 'pk', KeyType: 'HASH' },
+      { AttributeName: 'sk', KeyType: 'RANGE' }
+    ],
+    AttributeDefinitions: [
+      { AttributeName: 'pk', AttributeType: 'S' },
+      { AttributeName: 'sk', AttributeType: 'S' }
+    ],
+    ProvisionedThroughput: {
+      ReadCapacityUnits: 10000,
+      WriteCapacityUnits: 10000
```

Comment:
We should default to `PAY_PER_REQUEST`.

Reply:
Addressed in: [Commit](https://github.com/bcgov/parks-reso-api/pull/432/commits/7b4869559b502329cd1010b930eb54c80d7bc0a9)